Readability.js

```diff
@@ -217,19 +246,17 @@ Readability.prototype = {
     var doc = this._doc;
 
     // Remove all style tags in head
-    var styleTags = doc.getElementsByTagName("style");
-    for (var st = styleTags.length - 1; st >= 0; st -= 1) {
-      styleTags[st].parentNode.removeChild(styleTags[st]);
-    }
+    this._forEachNode(doc.getElementsByTagName("style"), function(styleNode) {
+      styleNode.parentNode.removeChild(styleNode);
+    });
```

Comment:
Hrmpf. I thought this was all going to go OK, and then I realized we'll still need to fix the `_setNodeTag` stuff for non-JSDOMParser, which will involve reparenting stuff (ie, below, move everything from a `font` tag into a different `span` tag and then replace the `font` with the `span`). This is unfortunate because if there are nested tags, we will be adding + removing things, which will mess up the loop. For instance:

`<font><font>Hello</font></font>`

We'll hit the first font first with the forEach loop, which then means we'll reparent the second one, which *might* mean we won't iterate over the second node? Not sure... :-\

Tested in Firefox's devtools console:
```
>>> document.getElementsByTagName("font").forEach
undefined
>>> Array.prototype.forEach.call(document.getElementsByTagName("font"), function(f) { var x = document.createElement("span"); Array.prototype.forEach.call(f.childNodes, function(fc) { x.appendChild(fc); }); f.parentNode.replaceChild(x, f); });
undefined
>>> document.getElementsByTagName("font")
HTMLCollection [ <font> ]
>>> document.getElementsByTagName("font")[0].parentNode
<span>
```